skops/card/_model_card.py

```diff
@@ -162,6 +164,44 @@ def metadata_from_config(config_path: Union[str, Path]) -> CardData:
     return card_data
 
 
+def _load_model(model: Any) -> Any:
+    """Loads the mddel if provided a file path, if already a model instance return it
+       unmodified.
+
+    Parameters
+    ----------
+    model : pathlib.path, str, or sklearn estimator
+        Path/str or the actual model instance. if a Path or str, loads the model on
+        first call.
+
+    Returns
+    -------
+    model : object
+        Model instance.
+
+    """
+
+    if not isinstance(model, (Path, str)):
+        return model
+
+    model_path = Path(model)
+    if not model_path.exists():
+        raise FileNotFoundError(f"File is not present:{model_path}")
+
+    if model_path.suffix in (".pkl", ".pickle"):
+        model = joblib.load(model_path)
+    elif model_path.suffix == ".skops":
+        model = load(model_path)
```

Comment:
we shouldn't rely on the extension here. We can check if the file is a zip file (`zipfile.is_zipfile(filename)`[¶](https://docs.python.org/3/library/zipfile.html#zipfile.is_zipfile) and load with `skops` if yes, otherwise try to load with `joblib`, and catch any exceptions which might be raised during each of them.

Reply:
can't we just keep a key in config file to store which format the model was serialized in?

Reply:
i agree depending on file extension is not a good way.
let's say the file loaded indeed was a zip then we will be conforming that it is sklearn estimator object. using this 
`[base.BaseEstimator](https://scikit-learn.org/stable/modules/generated/sklearn.base.BaseEstimator.html#sklearn.base.BaseEstimator)` ?

Reply:
> can't we just keep a key in config file to store which format the model was serialized in?

We could but I don't think this function is called late enough for that to work.

> let's say the file loaded indeed was a zip then we will be conforming that it is sklearn estimator object. using this
[base.BaseEstimator](https://scikit-learn.org/stable/modules/generated/sklearn.base.BaseEstimator.html#sklearn.base.BaseEstimator) ?

No, if it's a zip file, then it's a `skops` file, which means we can load using skops, otherwise we load with `joblib`, and we don't validate the output. We just catch any exception which might have been raised.